### DISCUSSION TITLE
DBus daemon: Managing requests

### DISCUSSION THREAD

**Discussion by o-alquimista** [brgl/libgpiod#116] [Q&A] [answered]
I'm using the DBus daemon through `gpiocli` from a shell script. The script is run by cron once a minute.

From my understanding, we're supposed to make a request and store its name for later reuse.

```sh
request=$(gpiocli request --output --chip="$chip" "$line")
# request0
gpiocli set --request="$request" "$line"=active
```

I can think of two different solutions:

1. Store it in a text file in `/var/run`
2. Extract it from the output of `gpiocli requests`, using the chip name and line number

The latter seems neater. Here's how I did it:

```sh
gpiocli requests
# request0 (gpiochip0) Offsets: [18]

request=$(gpiocli requests | grep "$chip" | grep "$line")
if [ $? -ne 0 ]; then
    # Make a new request
    gpiocli request --output --chip="$chip" "$line"
else
    # Reuse existing request
    request=$(echo "$request" | awk '{print $1}')
fi
```

Is this good? Or should I not be afraid of creating files in `/var/run`?

libgpiod v2.2

**Comment by brgl** [comment, marked as answer]
Yes, that's why `gpiocli requests` is there. Instead of parsing its output you may also use `gdbus` to inspect the properties of the request object directly if your system has it.
